feature_engine/imputation/drop_missing_data.py

```diff
@@ -12,26 +12,35 @@
 
 class DropMissingData(BaseImputer):
     """
-    The DropMissingData() will delete rows containing missing values. It works for both
-    numerical and categorical variables.
+    The DropMissingData() will delete rows containing missing values. It provides
+    similar functionality to pandas.drop_na().
 
-    You can enter the list of variables for which missing values should be removed from
-    the dataframe. Alternatively, the imputer will automatically select all variables
-    in the dataframe.
+    It works for both numerical and categorical variables. You can enter the list of
+    variables for which missing values should be removed from the dataframe.
+    Alternatively, the imputer will automatically select all variables in the dataframe.
 
-    More details in the :ref:`User Guide <drop_missing_data>`.
+    **Note**
+    The transformer will first select all variables or all user entered
+    variables and if `missing_only=True`, it will re-select from the original group
+    only those that show missing data in during fit, that is in the train set.
 
     Parameters
     ----------
     missing_only: bool, default=True
         If true, missing observations will be dropped only for the variables that have
-        missing data in the train set. If False, observations with NA will be dropped
-        from all variables.
+        missing data in the train set, during fit. If False, observations with NA
+        will be dropped from all variables indicated by the user.
 
     variables: list, default=None
         The list of variables to be imputed. If None, the imputer will find and
         select all variables in the dataframe.
 
+    drop_pct: float, default=None
+        The maximum percentage of data that can be missing in a row to be kept, i.e., if
```

Comment:
I spun on this for quite some time as well with myself. I again changed the name to `row_drop_pct` to try and be more explicit. The reason is because the class is called `DropMissingData` so when I personally think of `row_drop_pct` I think "This is the amount of data in a row that must be missing to drop the row." When I read myself an example "If I set `row_drop_pct` to 90% I think that 90% of the row must be missing to be dropped."

I completely see your point-of-view in that intuitively setting a number higher when it comes to dropping feels like more data should be dropped.
I personally don't see it this way as readily because I of the class name being DropMissingData. Because `thresh_pct` really becomes "this much data must be present in order to be kept." So then I found myself saying, "Why am I talking about the amount of data that must be kept if this class is all about dropping."

You're the expert here and I believe you have better intuition about how the larger community might read this parameter and will go back to naming it `thresh_pct` with the docs summarizing "amount of column data that must be present in order to keep a row."

Reply:
Regarding this statement:
We use isort to sort the import, black to style the code flake8 to test that it complies and mypy for typehints. Would you mind running those on the modified scripts to make sure they pass those tests?

Does tox not automatically take care of checking these? Sorry I remember from the ML Deployments course that I thought Chris mentioned black will automatically re-style incorrect text during a push as well. If not then do I need to individually run similar to mypy? `$ mypy drop_missing_data.py`? Also I've noticed tox complains about my lack of py 3.6, 3.7, 3.8 to check against. I set up a fresh conda env for this project that has 3.9. Can I add additional interpreters to an environment or what is the best way to resolve this?

Reply:
PS anytime I run `tox` it automatically wants to reformat the match_columns to match black, but you didn't want this, correct?
```
lint run-test: commands[1] | black feature_engine tests
reformatted feature_engine/preprocessing/match_columns.py
reformatted tests/test_preprocessing/test_match_columns.py
All done! ✨ 🍰 ✨
2 files reformatted, 127 files left unchanged.
```

Reply:
If you run tox on the entire package, it may modify all files (it should not though). So I would recommend running black and then flake8 just on the files you are updating.

Regarding mypy, it needs to be run on the entire project otherwise it does not find imports.
`mypy feature_engine`

Note that we do not use typehints in the tests, so no need to check those. But we do use black and flake8 for tests as well.